js/slidedeck.js

```diff
@@ -0,0 +1,394 @@
+      chart.style.display = 'block';
+      chart.src = slide.getAttribute('data-image-src');
+    } else {
+      // Hide the image and show the map for other slides
+      this.map.getContainer().style.display = 'block';
+      const image = document.getElementById('slide-image');
+      image.style.display = 'none';
+      const collection = await this.getSlideFeatureCollection(slide);
+      const layer = this.updateDataLayer(collection, slide.id);
+      /**
+     * Create a bounds object from a GeoJSON bbox array.
+     * @param {Array} bbox The bounding box of the collection
+     * @return {L.latLngBounds} The bounds object
+     */
+      const boundsFromBbox = (bbox) => {
+        const [west, south, east, north] = bbox;
+        const bounds = L.latLngBounds(
+            L.latLng(south, west),
+            L.latLng(north, east),
+        );
+        return bounds;
+      };
+      /**
+     * Create a temporary event handler that will show tooltips on the map
+     * features, after the map is done "flying" to contain the data layer.
+     */
+      const handleFlyEnd = () => {
+        if (slide.showpopups) {
+          layer.eachLayer((l) => {
+            l.bindTooltip(l.feature.properties.label, { permanent: true });
+            l.openTooltip();
+          });
+        }
+        this.map.removeEventListener('moveend', handleFlyEnd);
+      };
+
+      this.map.addEventListener('moveend', handleFlyEnd);
+      if (collection.bbox) {
+        this.map.flyToBounds(boundsFromBbox(collection.bbox));
+      } else {
+        this.map.flyToBounds(layer.getBounds());
+      }
+    }
+  }
+
+  /**
+   * Show the slide with ID matched by currentSlideIndex. If currentSlideIndex is
+   * null, then show the first slide.
+   */
+  syncMapToCurrentSlide() {
+    const slide = this.slides[this.currentSlideIndex];
+    this.syncMapToSlide(slide);
+  }
+
+  /**
+   * Increment the currentSlideIndex and show the corresponding slide. If the
+   * current slide is the final slide, then the next is the first.
+   */
+  goNextSlide() {
+    this.currentSlideIndex++;
+
+    if (this.currentSlideIndex === this.slides.length) {
+      this.currentSlideIndex = 0;
+    }
+
+    this.syncMapToCurrentSlide();
+  }
+
+  /**
+   * Decrement the currentSlideIndes and show the corresponding slide. If the
+   * current slide is the first slide, then the previous is the final.
+   */
+  goPrevSlide() {
+    this.currentSlideIndex--;
+
+    if (this.currentSlideIndex < 0) {
+      this.currentSlideIndex = this.slides.length - 1;
+    }
+
+    this.syncMapToCurrentSlide();
+  }
+
+  /**
+   * ### preloadFeatureCollections
+   *
+   * Initiate a fetch on all slide data so that the browser can cache the
+   * requests. This way, when a specific slide is loaded it has a better chance
+   * of loading quickly.
+   */
+  preloadFeatureCollections() {
+    for (const slide of this.slides) {
+      this.getSlideFeatureCollection(slide);
```

Comment:
**suggestion:** There are a bunch of 404 and JSON parsing errors in the console when the page loads. It's because one this `preloadFeatureCollections` function will run when the page loads to make later data requests faster, but not every slide has data associated with it. You might want to add the conditions you added in the `updateDataLayer` function here too. Something like:
```suggestion
      if (!slide.id.includes('image') && slide.id != 'chart-slide') {
        this.getSlideFeatureCollection(slide);
      }
```